### PR TITLE
Added Application Gateway Availability Zone Rule

### LINKED ISSUE #928: App Gateway should use availability zones for regions that support it
# Rule request

## Suggested rule change

If availability zones are available in the Application Gateway region, we should deploy an AZ SKU.

[Use zone aware services](https://docs.microsoft.com/azure/architecture/framework/resiliency/design-best-practices#use-zone-aware-services)

## Applies to the following

The rule applies to the following:

- Resource type: **Microsoft.Network/applicationGateways**

## Additional context

Related to #880.

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Fix #928

Created a `Azure.AppGw.AvailabilityZone` rule for the following types:
- Microsoft.Network/applicationGateways

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BernieWhite** [comment]
@ArmaanMcleod could be a regression with PSRule v1.7.x are you running v1.7.1? If you are try v1.6.0.

**Comment by ArmaanMcleod** [comment]
> @ArmaanMcleod could be a regression with PSRule v1.7.x are you running v1.7.1? If you are try v1.6.0.

Ah that seems to fix it. Thanks @BernieWhite.

File: docs/setup/configuring-rules.md [hunk 1]
```diff
@@ -264,4 +264,52 @@ Example:
 # YAML: Set the AZURE_AKS_ENABLED_PLATFORM_LOG_CATEGORIES_LIST configuration option to cluster-autoscaler and AllMetrics categories only. 
 configuration:
   AZURE_AKS_ENABLED_PLATFORM_LOG_CATEGORIES_LIST: ['cluster-autoscaler', 'AllMetrics']
-```
\ No newline at end of file
+```
+
+### Azure Application gateway additional region availability zone list
```
Comment (ArmaanMcleod) [inline docs/setup/configuring-rules.md:269]: Basically repeating the same example as `Azure.AKS.AvailabilityZone`. Perhaps we could condense the exampels into one?

Comment (BernieWhite) [reply docs/setup/configuring-rules.md:269]: Generally I would suggest separate rule configuration, but in this case it probably make sense to consolidate them.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Nice! All good to merge or add the config doc consolidation.

**Comment by BernieWhite** [comment]
> @ArmaanMcleod Armaan Dhaliwal-McLeod FTE could be a regression with PSRule v1.7.x are you running v1.7.1? If you are try v1.6.0.

Did v1.7.1 resolve the issue? Or is there an untracked bug that we need to address in v1.7.x?

**Comment by ArmaanMcleod** [comment]
> > @ArmaanMcleod Armaan Dhaliwal-McLeod FTE could be a regression with PSRule v1.7.x are you running v1.7.1? If you are try v1.6.0.
> 
> Did v1.7.1 resolve the issue? Or is there an untracked bug that we need to address in v1.7.x?

Yeah this was resolved by downgrading my `PSRule` from 1.7.1 to 1.6.0. I can probably create an issue about this. Basically only `docs/en/rules/module.md ` and `docs/en/rules/resource.md` were only being generated with 1.7.1.

File: docs/setup/configuring-rules.md (hunk 1 above)
Comment (ArmaanMcleod) [reply docs/setup/configuring-rules.md]: No worries @BernieWhite. I've consolidated the AZ rules. Let me know if your fine with it or want to make any changes 🙂.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Nice work! All good to merge.

**Pull request merged**